src/main/java/com/pinterest/secor/uploader/Uploader.java

```diff
@@ -233,7 +231,7 @@ private void trim(LogFilePath srcPath, long startOffset) throws Exception {
                         dstPath = new LogFilePath(localPrefix, srcPath.getTopic(),
                                                   srcPath.getPartitions(), srcPath.getGeneration(),
                                                   srcPath.getKafkaPartition(), startOffset,
-                                                  extension);
+                                                  srcPath.getExtension());
```

Comment:
In the case that file is a.snappy.parquet, srcPath.extension =  parquet, where is '.snappy' part being captured in LogFilePath?

Or how you do have 'a.snappy.parquet' in the first place, did you specify that in SecorConfig.getFileExtension config?

Reply:
We're using `secor.file.extension` config and specifying file extension. There is no other way to have double extension.
We are doing that to mimic Apache Spark file name convention, at least extensions.

MessageWriter is respecting that:
```java
if (mConfig.getCompressionCodec() != null && !mConfig.getCompressionCodec().isEmpty()) {
            mCodec = CompressionUtil.createCompressionCodec(mConfig.getCompressionCodec());
            mFileExtension = mCodec.getDefaultExtension();
        }
        if (mConfig.getFileExtension() != null && !mConfig.getFileExtension().isEmpty()) {
            mFileExtension = mConfig.getFileExtension();
        } else if (mFileExtension == null){
            mFileExtension = "";
        }
```
So I changed it to reuse same extention from path, generated by MessageWritter